modules/terraform-aws-bastion/main.tf

```diff
@@ -0,0 +1,103 @@
+# Copyright 2025 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ------- Elastic IP Allocation -------
+resource "aws_eip" "bastion_eip" {
+  count = var.create_eip ? 1 : 0
+
+  domain = "vpc"
+  tags   = merge(var.env_tags, { Name = var.eip_name })
+}
+
+resource "aws_instance" "bastion" {
+  ami                         = local.bastion_aws_ami
+  instance_type               = var.bastion_aws_instance_type
+  subnet_id                   = var.bastion_subnet_id
+  key_name                    = var.bastion_aws_keypair_name
+  vpc_security_group_ids      = [local.bastion_security_group_id]
+  associate_public_ip_address = var.enable_bastion_public_ip
+  user_data                   = var.bastion_user_data
+  user_data_replace_on_change = var.replace_on_user_data_change
+
+  tags = merge(var.env_tags, { Name = var.bastion_host_name })
+
+  availability_zone                    = var.bastion_az
+  iam_instance_profile                 = var.bastion_inst_profile
+  private_ip                           = var.bastion_private_ip
+  disable_api_termination              = var.disable_api_termination
+  instance_initiated_shutdown_behavior = var.bastion_shutdown_behaviour
+  source_dest_check                    = var.bastion_src_dest_check
+  monitoring                           = var.bastion_monitoring
+  tenancy                              = var.bastion_tenancy
+  placement_group                      = var.bastion_placement_grp
+  cpu_options {
+    core_count       = var.bastion_cpu_options != null ? var.bastion_cpu_options.core_count : null
```

Comment:
A slightly cleaner way to do this might be:
```
cpu_options {
  core_count = try(var.bastion_cpu_options.core_count, null) 
  threads_per_core = try(var.bastion_cpu_options.threads_per_core, null) 
```